src/core/io/src/4C_io_input_spec_builders.cpp

```diff
@@ -1229,6 +1229,51 @@ namespace
     return flattened_specs;
   }
 
+  // Convert an InputSpec of the form all_of(x,y, one_of(a,b,c))
+  // into one_of(all_of(x,y,a), all_of(x,y,b), all_of(x,y,c)).
+  std::vector<Core::IO::InputSpec> push_all_of_into_one_of(std::vector<Core::IO::InputSpec> specs)
```

Comment:
@gilrrei and I have been talking about this function for a longer time. It proofed to be very useful to tackle another subtle bug that may happen in certain, poorly written InputSpecs lacking substructure. In particular, this occurred for the `FUNCTION`s.